src/utils/AssayGeneratorMetabo.cpp

```diff
@@ -453,26 +453,10 @@ class TOPPAssayGeneratorMetabo :
             OPENMS_LOG_INFO << "Sirius Workspace could not be copied to " << sirius_workspace_directory << ". Please run AssayGeneratorMetabo with debug >= 2." << std::endl;
           }
         }
-       
-        // clean tmp directory if debug level < 2 
-        if (debug_level_ >= 2)
-        {
-          writeDebug_("Keeping temporary files in directory '" + tmp_dir + " and msfile at this location "+ tmp_ms_file + ". Set debug level to 1 or lower to remove them.", 2);
-        }
-        else
-        {
-          if (tmp_dir.empty() == false)
-          {
-            writeDebug_("Deleting temporary directory '" + tmp_dir + "'. Set debug level to 2 or higher to keep it.", 0);
-            File::removeDir(tmp_dir.toQString());
-          }
-          if (tmp_ms_file.empty() == false)
-          {
-            writeDebug_("Deleting temporary msfile '" + tmp_ms_file + "'. Set debug level to 2 or higher to keep it.", 0);
-            File::remove(tmp_ms_file); // remove msfile
-          }
-        }
 
+        // remove temporary SiriusAdapter file structure
+        SiriusAdapterAlgorithm::removeSiriusTmp(debug_level_, tmp_dir, tmp_ms_file);
```

Comment:
seems like we need to remember to call this function at every premature exit point?!
This sounds like it should be delegated to the compiler using RAII?! (i.e. create a small class, which calls this function in its destructor)

Reply:
Ah, did not know about that one - I will have a look. Any suggestions for a somewhat easily digestible (c++) programming concept read?